pkg/clients/grpc/connect.go

```diff
@@ -49,14 +49,22 @@ var (
 	errFailedToLoadRootCA        = errors.New("failed to load root ca file")
 )
 
-type Config struct {
+type BaseConfig struct {
+	URL          string        `env:"URL"             envDefault:"localhost:7001"`
+	Timeout      time.Duration `env:"TIMEOUT"         envDefault:"60s"`
 	ClientCert   string        `env:"CLIENT_CERT"     envDefault:""`
 	ClientKey    string        `env:"CLIENT_KEY"      envDefault:""`
 	ServerCAFile string        `env:"SERVER_CA_CERTS" envDefault:""`
-	URL          string        `env:"URL"             envDefault:"localhost:7001"`
-	Timeout      time.Duration `env:"TIMEOUT"         envDefault:"60s"`
-	AttestedTLS  bool          `env:"ATTESTED_TLS"    envDefault:"false"`
-	BackendInfo  string        `env:"BACKEND_INFO"    envDefault:""`
+	CertFile     string        `env:"SERVER_CERT"        envDefault:""`
+	KeyFile      string        `env:"SERVER_KEY"         envDefault:""`
+	ClientCAFile string        `env:"CLIENT_CA_CERTS"    envDefault:""`
+}
+
+type ClientConfig struct {
+	BaseConfig
+	BackendInfo string `env:"BACKEND_INFO"    envDefault:""`
+	ClientTLS   bool   `env:"CLIENT_TLS"    envDefault:"false"`
+	AttestedTLS bool   `env:"ATTESTED_TLS"       envDefault:"false"`
```

Comment:
backendinfo and attested tls are only for agent client(CLI). This needs to bbe  distinguished. Use interfaces for configuration purposes. clienttls is common for all clients and should be part of base config